wasm/src/error.rs

```diff
@@ -0,0 +1,58 @@
+use leb128::read;
+
+pub type Result<T> = std::result::Result<T, crate::error::Error>;
+
+#[repr(u8)]
+#[derive(Debug)]
+pub enum Error {
+    IoError(std::io::Error),
+    ReadZeroBytes,
+    TooManyBytes { expected: u8, found: u8 },
+    InvalidSectionOrder,
+    UnknownSectionID,
+    InvalidValueKind,
+    InvalidExternalKind,
+    InvalidSignatureType,
+    ArrayTooLarge,
+}
+
+impl std::fmt::Display for Error {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::IoError(e) => e.fmt(f)?,
+            Self::ReadZeroBytes => f.write_str("Tried to read from reader but got 0 bytes")?,
+            Self::TooManyBytes { expected, found } => f.write_str(&format!(
+                "Varint is too long! Expected {expected} bytes but found {found}"
+            ))?,
```

Comment:
Could avoid the allocation
```suggestion
            Self::TooManyBytes { expected, found } => write!(f, "Varint is too long! Expected {expected} bytes but found {found}")?,
```